planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -262,8 +262,26 @@ bool planning_scene_monitor::CurrentStateMonitor::haveCompleteState(const ros::D
   return result;
 }
 
+bool planning_scene_monitor::CurrentStateMonitor::waitForCurrentState(const ros::Time t, double wait_time) const
+{
+  ros::WallTime start = ros::WallTime::now();
+  ros::WallDuration elapsed;
+  ros::WallDuration timeout(wait_time);
+
+  boost::mutex::scoped_lock lock(state_update_lock_);
```

Comment:
Looking into this a bit I don't see how this function could ever return without timeout.
`jointStateCallback` requires the lock for a successful update of `current_state_time_` et al.
I would propose to use `getCurrentStateTime()` in the condition below and remove the lock.